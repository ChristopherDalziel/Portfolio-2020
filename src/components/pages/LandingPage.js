import React from "react";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Final_Col.jpg";
import mobileBackgroundImage from "../../assets/Web_Final_Col_Mob.jpg";
import { keyframes } from "@emotion/core";

const fadeInAnimation = keyframes`
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}`;

const LandingPageContainer = styled.main`
  display: flex;
  flex-direction: column;
  background-image: url(${backgroundImage});
  background-size: cover;
  height: 100vh;
  width: 100vw;
  @media (max-width: 815px) {
    background-image: url(${mobileBackgroundImage});
    background-size: auto 80vh;
    background-repeat: no-repeat;
`;

const HeroTextContainer = styled.div`
  animation: ${fadeInAnimation} ease 3s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  height: 80%;
  display: flex;
  flex-direction: column;
  background-color: transparent;
  color: #493a37;
  margin-left: 18%;
`;

const Heading = styled.h1`
  margin-top: 30%;
  font-size: 6em;
  @media (max-width: 815px) {
    margin-top: 60vh;
    font-size: 60px;
`;

const SubHeading = styled.h2`
  font-size: 3em;
  @media (max-width: 815px) {
    font-size: 25px;
`;

function LandingPage() {
  return (
    <LandingPageContainer>
      <HeroTextContainer>
        <Heading>HELLO.</Heading>
        <SubHeading>my name's Chris.</SubHeading>
      </HeroTextContainer>
    </LandingPageContainer>
  );
}

export default LandingPage;
