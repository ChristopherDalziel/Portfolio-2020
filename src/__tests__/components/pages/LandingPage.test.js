import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import LandingPage from "../../../components/pages/LandingPage";

test("Should render Landing Page correctly", () => {
  const wrapper = shallow(<LandingPage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
