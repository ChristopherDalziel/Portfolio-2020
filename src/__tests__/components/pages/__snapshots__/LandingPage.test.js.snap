// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Landing Page correctly 1`] = `
<Styled(main)>
  <Styled(div)>
    <Styled(h1)>
      HELLO.
    </Styled(h1)>
    <Styled(h2)>
      my name's Chris.
    </Styled(h2)>
  </Styled(div)>
</Styled(main)>
`;
